File: Reproduced/RQ1/perl/benchmark.pl
use Time::HiRes qw(gettimeofday);

sub measure {
    my ($data, $pattern) = @_;

    my $start = Time::HiRes::gettimeofday();

    my $count = () = $data =~ /$pattern/g;

    my $elapsed = (Time::HiRes::gettimeofday() - $start) * 1e3;

    printf("%f - %d\n", $elapsed, $count);
}

# sub measure {
#   my ($data, $pattern) = @_;

#   my $start = Time::HiRes::gettimeofday();

#   # 修改这行，使用 =~ m/^pattern$/ 进行完全匹配
#   my $match = $data =~ /^($pattern)$/;

#   my $elapsed = (Time::HiRes::gettimeofday() - $start) * 1e3;

#   # 修改输出，显示是否匹配（1表示匹配，空或0表示不匹配）
#   printf("%f - %d\n", $elapsed, $match ? 1 : 0);
# }

if (@ARGV != 2) {
  die "Usage: ./benchmark.pl <filename>\n";
}

my ($filename) = @ARGV[1];

open my $fh, '<', $filename or die 'Could not open file.';
my $text;
read $fh, $data, -s $filename;
close $fh;
measure $data, @ARGV[0];

File: Reproduced/RQ1/python/benchmark.py
import sys
import re
from timeit import default_timer as timer

if len(sys.argv) != 3:
    print('Usage: python benchmark.py <filename>')
    sys.exit(1)

def measure(data, pattern):
    start_time = timer()
    matches = re.findall(pattern, data)

    elapsed_time = timer() - start_time

    print(str(elapsed_time * 1e3))

# def measure(data, pattern):
#     start_time = timer()
#     match = re.fullmatch("^(" + pattern + ")$", data)

#     elapsed_time = timer() - start_time

#     print(str(elapsed_time * 1e3))

#f1 = open(sys.argv[2], encoding='utf-8')
f1 = open(sys.argv[2], "rb")
Lines = f1.read()
str1 = ""
for c in Lines:
    str1 = str1 + chr(c)
measure(str1, sys.argv[1])